Tests/Contour.Configurator.Tests/BusConfigurationSpecs.cs

```diff
@@ -1078,6 +1079,150 @@ public void should_route_any()
                 }
             }
         }
+
+        [TestFixture]
+        [Category("Unit")]
+        public class when_configuring_endpoint_incoming
+        {
+            [Test]
+            public void should_set_qos_if_present()
+            {
+                const string endpointName = "ep";
+                const int prefetchCount = 5;
+                const int prefetchSize = 6;
+                const string onKeyName = "key";
+
+                string config =
+                    $@"<endpoints>
+                        <endpoint name=""{endpointName}"" connectionString=""amqp://localhost:666"">
+                            <qos prefetchCount=""123"" prefetchSize=""456"" />
+                                <incoming>
+                                    <on key=""{onKeyName}"" label=""msg.a"" react=""DynamicHandler"" requiresAccept=""true"">
+                                        <qos prefetchCount=""{prefetchCount}"" prefetchSize=""{prefetchSize}"" />
+                                    </on>
+                                </incoming>
+                        </endpoint>
+                    </endpoints>";
+
+                var resoverMock = new Mock<IDependencyResolver>();
+                var section = new XmlEndpointsSection(config);
+                var configurator = new AppConfigConfigurator(section, resoverMock.Object);
+                
+                var busConfiguration = new BusConfiguration();
+                busConfiguration.UseRabbitMq(); //Basic receiver configurator and receiver options are actually unaware of any QoS settings; so these tests are not really Contour specific
+
+                var result = configurator.Configure(endpointName, busConfiguration);
+                var busConfigurationResult = (BusConfiguration) result;
+                var receiverConfiguration = busConfigurationResult.ReceiverConfigurations.First();
+                var receiverOptions = (RabbitReceiverOptions) receiverConfiguration.Options;
+                
+                var qos = receiverOptions.GetQoS();
+                var value = qos.Value;
+
+                value.PrefetchCount.Should().Be(prefetchCount, "Incoming QoS prefetch count should be set");

```

Comment:
As discussed we need to split the test in two to check prefetch count and prefetch size separatelly.